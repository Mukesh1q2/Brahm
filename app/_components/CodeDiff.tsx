"use client";

import React from 'react';
import dynamic from 'next/dynamic';

export type CodeDiffProps = {
  original?: string;
  modified: string;
  language?: string;
  height?: string;
};

// Wrap DiffEditor from @monaco-editor/react with dynamic import (client-only), with typed props
const Diff = dynamic<CodeDiffProps>(
  async () => {
    const mod: any = await import('@monaco-editor/react');
    const DiffEditor = (mod as any).DiffEditor || (mod as any).default;
    const Comp: React.FC<CodeDiffProps> = (props) => (
      <DiffEditor
        height={props.height || '280px'}
        language={props.language || 'plaintext'}
        original={props.original || ''}
        modified={props.modified || ''}
        theme="vs-dark"
        options={{ readOnly: true, renderSideBySide: true, minimap: { enabled: false } }}
      />
    );
    return Comp as any;
  },
  { ssr: false }
);

export default function CodeDiff({ original, modified, language, height }: CodeDiffProps) {
  return <Diff original={original} modified={modified} language={language} height={height} />;
}

