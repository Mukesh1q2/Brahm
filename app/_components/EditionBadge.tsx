"use client";

import React from 'react';
import { useEdition } from '@/store/edition';

export default function EditionBadge() {
  const { edition, setEdition } = useEdition();
  const [mounted, setMounted] = React.useState(false);
  React.useEffect(() => setMounted(true), []);

  const toggle = async () => {
    const next = edition === 'advanced' ? 'basic' : 'advanced';
    try {
      await fetch('/api/edition', { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify({ edition: next }) });
    } catch {}
    setEdition(next);
  };

  // Avoid SSR mismatch; render after mount
  if (!mounted) return null;

  return (
    <button
      type="button"
      onClick={toggle}
      title={`Current: ${edition}. Click to switch.`}
      className="fixed bottom-4 right-4 z-[1000] inline-flex items-center gap-2 rounded-full border border-white/10 bg-white/5 px-3 py-1.5 text-xs text-white/80 hover:bg-white/10 backdrop-blur"
      aria-label={`Edition ${edition}`}
    >
      <span className="h-2 w-2 rounded-full"
        style={{ backgroundColor: edition === 'advanced' ? '#7c3aed' : '#64748b' }}
      />
      <span>Edition: <strong className="ml-1 capitalize">{edition}</strong></span>
      <span className="ml-2 rounded bg-white/10 px-2 py-0.5 text-[11px]">Switch</span>
    </button>
  );
}

