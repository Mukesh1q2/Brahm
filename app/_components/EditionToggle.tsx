"use client";

import React from 'react';
import { useEdition } from '@/store/edition';

export default function EditionToggle({ className = '' }: { className?: string }) {
  const { edition, setEdition } = useEdition();
  const [mounted, setMounted] = React.useState(false);

  React.useEffect(() => {
    setMounted(true);
    // Ensure html has a class for CSS hooks on mount
    try {
      const root = document.documentElement;
      root.classList.remove('edition-basic','edition-advanced');
      root.classList.add(edition === 'advanced' ? 'edition-advanced' : 'edition-basic');
    } catch {}
  }, [edition]);

  // Hydrate from localStorage after mount to avoid hydration mismatch
  React.useEffect(() => {
    try {
      const saved = localStorage.getItem('brahm:edition') as 'basic' | 'advanced' | null;
      if (saved && saved !== edition) setEdition(saved);
    } catch {}
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div className={`inline-flex items-center gap-1 text-xs ${className}`} suppressHydrationWarning>
      <span className="text-gray-400">Edition</span>
      <div className="inline-flex rounded-md overflow-hidden border border-white/10">
        <button
          type="button"
          onClick={async () => { try { await fetch('/api/edition', { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify({ edition: 'basic' }) }); } catch {} setEdition('basic'); }}
          className={`px-2 py-1 ${(mounted && edition==='basic') ? 'bg-white/15 text-white' : 'bg-white/5 text-white/70 hover:bg-white/10'}`}
          aria-pressed={edition==='basic'}
          title="Switch to Basic"
        >Basic</button>
        <button
          type="button"
          onClick={async () => { try { await fetch('/api/edition', { method: 'POST', headers: { 'Content-Type': 'application/json' }, body: JSON.stringify({ edition: 'advanced' }) }); } catch {} setEdition('advanced'); }}
          className={`px-2 py-1 ${(mounted && edition==='advanced') ? 'bg-brand-600/60 text-white' : 'bg-white/5 text-white/70 hover:bg-white/10'}`}
          aria-pressed={edition==='advanced'}
          title="Switch to Advanced"
        >Advanced</button>
      </div>
    </div>
  );
}
