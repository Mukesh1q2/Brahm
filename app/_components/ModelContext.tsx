"use client";

import React from "react";

export const modelOptions = [
  "auto",
  "gpt-4o",
  "gpt-4o-mini",
  "gpt-3.5-turbo",
  "claude-3-5-sonnet",
  "gemini-2.5-pro",
  "local-llama",
] as const;

export const ratePer1k: Record<string, number> = {
  "auto": 0.0, // router decides; cost computed server-side
  "gpt-4o": 0.005,
  "gpt-4o-mini": 0.0006,
  "gpt-3.5-turbo": 0.0015,
  "claude-3-5-sonnet": 0.005, // placeholder
  "gemini-2.5-pro": 0.004, // placeholder estimate
  "local-llama": 0.0,
};

// API base for fetch interception (Panini endpoints)
const API_BASE = process.env.NEXT_PUBLIC_API_BASE || process.env.NEXT_PUBLIC_API_URL || "http://localhost:8000";

export type ModelContextValue = {
  model: string;
  setModel: (m: string) => void;
  options: readonly string[];
  rates: Record<string, number>;
  expert: boolean;
  setExpert: (b: boolean) => void;
  globalModelSelectorEnabled: boolean;
  // budgets & region
  region: string | null;
  setRegion: (r: string | null) => void;
  budgetUSDMonthly: number | null;
  spentUSDMonthToDate: number;
  setBudgetUSDMonthly: (v: number | null) => void;
  resetSpent: () => void;
};

const Ctx = React.createContext<ModelContextValue | null>(null);

export function ModelProvider({ children }: { children: React.ReactNode }) {
  const [model, setModelState] = React.useState<string>('auto');
  const [expert, setExpertState] = React.useState<boolean>(false);
  const [region, setRegionState] = React.useState<string | null>(null);
  const [budgetUSDMonthly, setBudgetUSDMonthlyState] = React.useState<number | null>(null);
  const [spentUSDMonthToDate, setSpent] = React.useState<number>(0);
  // Hydrate from localStorage after mount to avoid SSR/client mismatch
  React.useEffect(() => {
    try {
      const m = window.localStorage.getItem('model_selected');
      if (m) setModelState(m);
    } catch {}
    try {
      const ex = window.localStorage.getItem('expert_mode') === 'true';
      if (ex) setExpertState(true);
    } catch {}
    try {
      const r = window.localStorage.getItem('region_selected');
      setRegionState(r || null);
    } catch {}
    try {
      const b = window.localStorage.getItem('budget_usd_monthly');
      setBudgetUSDMonthlyState(b ? Number(b) : null);
    } catch {}
    try {
      const s = window.localStorage.getItem('spent_usd_mtd');
      setSpent(s ? Number(s) : 0);
    } catch {}
  }, []);
  const setExpert = React.useCallback((b: boolean) => {
    setExpertState(b);
    try { window.localStorage.setItem('expert_mode', String(b)); } catch {}
  }, []);
  const setModel = React.useCallback((m: string) => {
    setModelState(m);
    try { window.localStorage.setItem("model_selected", m); } catch {}
  }, []);
  const setRegion = React.useCallback((r: string | null) => {
    setRegionState(r);
    try { if (r) window.localStorage.setItem('region_selected', r); else window.localStorage.removeItem('region_selected'); } catch {}
  }, []);
  const setBudgetUSDMonthly = React.useCallback((v: number | null) => {
    setBudgetUSDMonthlyState(v);
    try { if (v != null) window.localStorage.setItem('budget_usd_monthly', String(v)); else window.localStorage.removeItem('budget_usd_monthly'); } catch {}
  }, []);
  const resetSpent = React.useCallback(() => {
    setSpent(0);
    try { window.localStorage.setItem('spent_usd_mtd', '0'); } catch {}
  }, []);
  const globalModelSelectorEnabled = (process.env.NEXT_PUBLIC_GLOBAL_MODEL_SELECTOR ?? 'true') !== 'false';
  const val = React.useMemo<ModelContextValue>(() => ({
    model,
    setModel,
    options: modelOptions as unknown as string[],
    rates: ratePer1k,
    expert,
    setExpert,
    globalModelSelectorEnabled,
    region,
    setRegion,
    budgetUSDMonthly,
    spentUSDMonthToDate: spentUSDMonthToDate,
    setBudgetUSDMonthly,
    resetSpent,
  }), [model, setModel, expert, setExpert, globalModelSelectorEnabled, region, setRegion, budgetUSDMonthly, spentUSDMonthToDate, setBudgetUSDMonthly, resetSpent]);

  // Global fetch interceptor for Panini endpoints to inject X-Model and basic client metadata
  const modelRef = React.useRef(model);
  React.useEffect(() => { modelRef.current = model; }, [model]);
  const origFetchRef = React.useRef<typeof fetch | null>(null);
  const patchedRef = React.useRef(false);

  // Attach fetch interceptor for Panini endpoints
  React.useEffect(() => {
    if (typeof window === 'undefined') return;
    if (patchedRef.current) return;
    const orig = window.fetch.bind(window);
    origFetchRef.current = orig;
    const base = (() => { try { return new URL(API_BASE, window.location.origin); } catch { return null; } })();
    const shouldPatch = (u: URL) => {
      const path = u.pathname || '';
      const isPaniniPath = path.startsWith('/panini') || path.startsWith('/optimize/panini');
      if (!isPaniniPath) return false;
      if (!base) return true;
      return (u.host === base.host && u.protocol === base.protocol);
    };
    window.fetch = async (input: any, init?: RequestInit) => {
      try {
        const urlStr = typeof input === 'string' || input instanceof URL ? String(input) : (input && input.url ? String(input.url) : '');
        const u = new URL(urlStr, typeof window !== 'undefined' ? window.location.origin : 'http://localhost');
        const start = Date.now();
        const trace = (typeof crypto !== 'undefined' && 'randomUUID' in crypto) ? (crypto as any).randomUUID() : `${start}-${Math.random().toString(36).slice(2)}`;
        const isPanini = shouldPatch(u);
        let headersObj: Record<string, string> = {};
        let hasXModel = false;
        if (init && init.headers) {
          if (init.headers instanceof Headers) {
            init.headers.forEach((v,k)=>{ headersObj[k] = v; if (k.toLowerCase() === 'x-model') hasXModel = true; });
          } else if (Array.isArray(init.headers)) {
            for (const [k,v] of init.headers) { headersObj[k] = String(v); if (k.toLowerCase() === 'x-model') hasXModel = true; }
          } else {
            const h = init.headers as any;
            for (const k in h) { if (Object.prototype.hasOwnProperty.call(h,k)) { headersObj[k] = String(h[k]); if (k.toLowerCase() === 'x-model') hasXModel = true; } }
          }
        } else if (input && input.headers) {
          try { (input.headers as Headers).forEach((v:string,k:string)=>{ headersObj[k] = v; if (k.toLowerCase() === 'x-model') hasXModel = true; }); } catch {}
        }
        if (isPanini) {
          if (!hasXModel) headersObj['X-Model'] = modelRef.current;
          headersObj['X-Client-App'] = 'panini';
          headersObj['X-Trace-Id'] = trace;
          headersObj['X-Request-Id'] = trace;
          headersObj['X-Client-Start'] = String(start);
          try { const r = window.localStorage.getItem('region_selected'); if (r) headersObj['X-Region'] = r; } catch {}
        }
        const newInit: RequestInit = { ...(init || {}), headers: headersObj };
        let res: Response;
        if (input instanceof Request) {
          const req = new Request(input, newInit);
          res = await orig(req as any);
        } else {
          res = await orig(String(u), newInit);
        }
        if (isPanini) {
          const end = Date.now();
          const clientLatency = end - start;
          try {
            const r = res.clone();
            const h = r.headers;
            const serverLatency = Number(h.get('x-server-latency-ms') || h.get('x-latency-ms') || h.get('x-elapsed-ms'));
            const costUsd = Number(h.get('x-llm-cost-usd'));
            const respModel = h.get('x-llm-model') || h.get('x-model') || '';
          const detail = { trace, url: u.toString(), ok: res.ok, status: res.status, clientLatencyMs: clientLatency, serverLatencyMs: Number.isFinite(serverLatency) ? serverLatency : null, costUsd: Number.isFinite(costUsd) ? costUsd : null, requestModel: modelRef.current, responseModel: respModel, app: 'panini' };
          window.dispatchEvent(new CustomEvent('telemetry:request', { detail }));
          } catch {}
        }
        return res;
      } catch (e) {
        return orig(input, init as any);
      }
    };
    patchedRef.current = true;
    return () => {
      if (origFetchRef.current) window.fetch = origFetchRef.current as any;
      patchedRef.current = false;
    };
  }, []);

  // Optional telemetry sink: POST live telemetry to an ingest endpoint if configured
  React.useEffect(() => {
    if (typeof window === 'undefined') return;
    const ingest = (process.env.NEXT_PUBLIC_TELEMETRY_INGEST || '').trim() || `${API_BASE}/telemetry/ingest`;
    if (!ingest) return;
    const handler = async (ev: any) => {
      try {
        const detail = ev?.detail || {};
        // Update local spend meter if provided
        const costUsd = Number(detail.costUsd ?? NaN);
        if (Number.isFinite(costUsd) && costUsd > 0) {
          setSpent(prev => {
            const next = prev + costUsd;
            try { window.localStorage.setItem('spent_usd_mtd', String(next)); } catch {}
            return next;
          });
        }
        const body = {
          ts: new Date().toISOString(),
          trace: detail.trace || '',
          url: detail.url || '',
          ok: !!detail.ok,
          status: Number(detail.status || 0),
          clientLatencyMs: detail.clientLatencyMs ?? null,
          serverLatencyMs: detail.serverLatencyMs ?? null,
          costUsd: detail.costUsd ?? null,
          requestModel: detail.requestModel || null,
          responseModel: detail.responseModel || null,
          client: 'frontend',
          app: detail.app || 'panini',
        };
        // include auth token if present (best effort)
        let headers: Record<string, string> = { 'Content-Type': 'application/json' };
        try { const t = window.localStorage.getItem('access_token'); if (t) headers['Authorization'] = `Bearer ${t}`; } catch {}
        await fetch(ingest, { method: 'POST', headers, body: JSON.stringify(body), keepalive: true });
      } catch {}
    };
    window.addEventListener('telemetry:request', handler as any);
    return () => window.removeEventListener('telemetry:request', handler as any);
  }, []);

  return <Ctx.Provider value={val}>{children}</Ctx.Provider>;
}

export function useModel(): ModelContextValue {
  const ctx = React.useContext(Ctx);
  if (!ctx) throw new Error("useModel must be used within ModelProvider");
  return ctx;
}

