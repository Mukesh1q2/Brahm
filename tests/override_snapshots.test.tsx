import React from 'react';
import { render, screen } from '@testing-library/react';
import EditorPage from '../app/editor/page';
import CanvasPage from '../app/canvas/page';
import ConsolePage from '../app/console/page';
import { ModelProvider } from '../app/_components/ModelContext';

function Provider({ children }: { children: React.ReactNode }) {
  return <ModelProvider>{children}</ModelProvider>;
}

describe('Override UI snapshots', () => {
  it('Editor shows "Using global" by default', () => {
    const { asFragment } = render(<Provider><EditorPage /></Provider>);
    expect(screen.getByText(/Using global:/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  it('Canvas shows "Using global" by default', () => {
    const { asFragment } = render(<Provider><CanvasPage /></Provider>);
    expect(screen.getByText(/Using global:/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  it('Console renders model filter', () => {
    const { asFragment } = render(<Provider><ConsolePage /></Provider>);
    // At least one combobox (model selector)
    expect(screen.getAllByRole('combobox').length).toBeGreaterThan(0);
    expect(asFragment()).toMatchSnapshot();
  });
});

