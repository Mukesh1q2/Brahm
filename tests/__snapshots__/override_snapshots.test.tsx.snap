// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Override UI snapshots Canvas shows "Using global" by default 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen bg-gray-950 text-gray-100"
  >
    <div
      class="p-4 border-b border-purple-500/20 flex items-center justify-between"
    >
      <div
        class="flex items-center gap-3"
      >
        <svg
          class="lucide lucide-terminal-square text-purple-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m7 11 2-2-2-2"
          />
          <path
            d="M11 13h4"
          />
          <rect
            height="18"
            rx="2"
            ry="2"
            width="18"
            x="3"
            y="3"
          />
        </svg>
        <h1
          class="text-xl font-semibold"
        >
          Brahm Canvas
        </h1>
        <span
          class="text-xs text-purple-300/70"
        >
          Jnana River
        </span>
      </div>
      <div
        class="flex items-center gap-2"
      >
        <label
          class="text-xs text-purple-300 flex items-center gap-2"
        >
          <span>
            Model
          </span>
          <select
            class="bg-gray-900 border border-gray-700 rounded px-2 py-1 text-gray-100 text-sm"
          >
            <option
              value="auto"
            >
              auto
            </option>
            <option
              value="gpt-4o"
            >
              gpt-4o
            </option>
            <option
              value="gpt-4o-mini"
            >
              gpt-4o-mini
            </option>
            <option
              value="gpt-3.5-turbo"
            >
              gpt-3.5-turbo
            </option>
            <option
              value="claude-3-5-sonnet"
            >
              claude-3-5-sonnet
            </option>
            <option
              value="local-llama"
            >
              local-llama
            </option>
          </select>
        </label>
        <label
          class="text-xs text-purple-300 flex items-center gap-1"
        >
          <input
            type="checkbox"
          />
           override
        </label>
        <button
          class="text-[11px] px-2 py-1 rounded bg-gray-800 border border-gray-700 hover:bg-gray-700"
          title="Revert to global model"
        >
          Use global
        </button>
        <span
          class="hidden md:inline text-xs text-purple-300/80"
        >
          Using global: 
          <b>
            auto
          </b>
        </span>
        <input
          class="bg-gray-800/70 rounded px-3 py-2 w-[380px] text-sm focus:outline-none focus:ring-2 focus:ring-purple-600/40"
          placeholder="Describe your goal..."
          value="Write a Python function that reverses a string."
        />
        <button
          class="px-4 py-2 rounded bg-purple-600 hover:bg-purple-700 disabled:opacity-50 flex items-center gap-2"
        >
          <svg
            class="lucide lucide-send "
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m22 2-7 20-4-9-9-4Z"
            />
            <path
              d="M22 2 11 13"
            />
          </svg>
           Run Plan
        </button>
      </div>
    </div>
    <div
      class="h-[calc(100vh-64px)]"
    >
      <div
        class="grid grid-cols-2 gap-0 h-full"
      >
        <div
          class="border-r border-purple-500/20"
        >
          <div
            class="p-2 text-xs bg-gray-900/40 border-b border-purple-500/20"
          >
            Drag & drop files here to OCR (stub)
          </div>
          <div
            data-testid="dynamic-stub"
          />
        </div>
        <div
          class="p-3"
        >
          <div
            class="flex items-center justify-between mb-2 text-[11px] text-gray-400"
          >
            <span>
              Output
            </span>
            <div
              class="text-[11px] text-gray-500 flex items-center gap-2"
            >
              <span
                class="hidden md:inline"
              >
                Requests:
              </span>
              <span
                class="px-2 py-0.5 rounded bg-gray-900 border border-gray-700"
              >
                0
              </span>
            </div>
          </div>
          <pre
            class="h-full overflow-auto bg-gray-900 rounded p-3 text-xs"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Override UI snapshots Console renders model filter 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen bg-gray-950 text-gray-100"
  >
    <header
      class="p-4 border-b border-purple-500/20 flex items-center justify-between"
    >
      <div
        class="flex items-center gap-3"
      >
        <svg
          class="lucide lucide-list-filter text-purple-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 6h18"
          />
          <path
            d="M7 12h10"
          />
          <path
            d="M10 18h4"
          />
        </svg>
        <h1
          class="text-xl font-semibold"
        >
          Self-Evolving Console
        </h1>
        <span
          class="text-xs text-purple-300/70"
        >
          Audit Filters
        </span>
      </div>
      <div
        class="flex items-center gap-3"
      >
        <label
          class="text-xs text-purple-300 flex items-center gap-2"
        >
          <span>
            Model
          </span>
          <select
            class="bg-gray-900 border border-gray-700 rounded px-2 py-1 text-gray-100 text-sm"
          >
            <option
              value="auto"
            >
              auto
            </option>
            <option
              value="gpt-4o"
            >
              gpt-4o
            </option>
            <option
              value="gpt-4o-mini"
            >
              gpt-4o-mini
            </option>
            <option
              value="gpt-3.5-turbo"
            >
              gpt-3.5-turbo
            </option>
            <option
              value="claude-3-5-sonnet"
            >
              claude-3-5-sonnet
            </option>
            <option
              value="local-llama"
            >
              local-llama
            </option>
          </select>
        </label>
        <button
          class="px-3 py-2 rounded bg-gray-800 hover:bg-gray-700 flex items-center gap-2 text-sm"
        >
          <svg
            class="lucide lucide-refresh-cw "
            fill="none"
            height="16"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 12a9 9 0 0 1 9-9 9.75 9.75 0 0 1 6.74 2.74L21 8"
            />
            <path
              d="M21 3v5h-5"
            />
            <path
              d="M21 12a9 9 0 0 1-9 9 9.75 9.75 0 0 1-6.74-2.74L3 16"
            />
            <path
              d="M8 16H3v5"
            />
          </svg>
           Refresh
        </button>
      </div>
    </header>
    <main
      class="max-w-6xl mx-auto p-4 space-y-4"
    >
      <div
        class="bg-black/30 border border-purple-500/20 rounded-lg p-4"
      >
        <div
          class="flex items-center justify-between mb-2"
        >
          <div
            class="flex items-center gap-3"
          >
            <div
              class="text-sm font-semibold text-purple-300"
            >
              Live Telemetry
            </div>
            <div
              class="text-xs text-gray-400"
            >
              session-only, from client-side fetch
            </div>
          </div>
          <div
            class="flex items-center gap-2"
          >
            <button
              class="px-2 py-1 rounded bg-gray-800 hover:bg-gray-700 text-xs"
            >
              Show (0)
            </button>
            <button
              class="px-2 py-1 rounded bg-gray-800 hover:bg-gray-700 text-xs"
            >
              Clear
            </button>
          </div>
        </div>
      </div>
      <div
        class="bg-black/30 border border-purple-500/20 rounded-lg p-4"
      >
        <div
          class="grid md:grid-cols-6 gap-3"
        >
          <div>
            <label
              class="text-xs text-purple-300"
            >
              Stage
            </label>
            <select
              class="w-full bg-gray-900 border border-gray-700 rounded px-2 py-1"
            >
              <option
                value=""
              >
                Any
              </option>
              <option
                value="plan"
              >
                plan
              </option>
              <option
                value="dry-run"
              >
                dry-run
              </option>
              <option
                value="shadow-run"
              >
                shadow-run
              </option>
              <option
                value="approve"
              >
                approve
              </option>
              <option
                value="execute"
              >
                execute
              </option>
            </select>
          </div>
          <div>
            <label
              class="text-xs text-purple-300"
            >
              Allowed
            </label>
            <select
              class="w-full bg-gray-900 border border-gray-700 rounded px-2 py-1"
            >
              <option
                value=""
              >
                Any
              </option>
              <option
                value="true"
              >
                true
              </option>
              <option
                value="false"
              >
                false
              </option>
            </select>
          </div>
          <div>
            <label
              class="text-xs text-purple-300"
            >
              User
            </label>
            <input
              class="w-full bg-gray-900 border border-gray-700 rounded px-2 py-1"
              value=""
            />
          </div>
          <div>
            <label
              class="text-xs text-purple-300"
            >
              Tool
            </label>
            <input
              class="w-full bg-gray-900 border border-gray-700 rounded px-2 py-1"
              value=""
            />
          </div>
          <div>
            <label
              class="text-xs text-purple-300"
            >
              Model
            </label>
            <select
              class="w-full bg-gray-900 border border-gray-700 rounded px-2 py-1"
            >
              <option
                value=""
              >
                Any
              </option>
              <option
                value="auto"
              >
                auto
              </option>
              <option
                value="gpt-4o"
              >
                gpt-4o
              </option>
              <option
                value="gpt-4o-mini"
              >
                gpt-4o-mini
              </option>
              <option
                value="gpt-3.5-turbo"
              >
                gpt-3.5-turbo
              </option>
              <option
                value="claude-3-5-sonnet"
              >
                claude-3-5-sonnet
              </option>
              <option
                value="local-llama"
              >
                local-llama
              </option>
            </select>
          </div>
          <div
            class="md:col-span-2"
          >
            <label
              class="text-xs text-purple-300"
            >
              Search
            </label>
            <div
              class="flex items-center gap-2"
            >
              <svg
                class="lucide lucide-search text-gray-400"
                fill="none"
                height="16"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
                <path
                  d="m21 21-4.3-4.3"
                />
              </svg>
              <input
                class="flex-1 bg-gray-900 border border-gray-700 rounded px-2 py-1"
                placeholder="command contains..."
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="text-xs text-purple-300"
            >
              Start (ISO)
            </label>
            <input
              class="w-full bg-gray-900 border border-gray-700 rounded px-2 py-1"
              placeholder="2025-08-21T00:00:00Z"
              value=""
            />
          </div>
          <div>
            <label
              class="text-xs text-purple-300"
            >
              End (ISO)
            </label>
            <input
              class="w-full bg-gray-900 border border-gray-700 rounded px-2 py-1"
              placeholder="2025-08-22T00:00:00Z"
              value=""
            />
          </div>
          <div>
            <label
              class="text-xs text-purple-300"
            >
              Order
            </label>
            <select
              class="w-full bg-gray-900 border border-gray-700 rounded px-2 py-1"
            >
              <option
                value="desc"
              >
                Newest
              </option>
              <option
                value="asc"
              >
                Oldest
              </option>
            </select>
          </div>
          <div>
            <label
              class="text-xs text-purple-300"
            >
              Page size
            </label>
            <input
              class="w-full bg-gray-900 border border-gray-700 rounded px-2 py-1"
              max="100"
              min="1"
              type="number"
              value="20"
            />
          </div>
          <div
            class="flex items-end"
          >
            <button
              class="px-3 py-2 rounded bg-purple-700 hover:bg-purple-800 text-sm"
            >
              Apply
            </button>
          </div>
        </div>
      </div>
      <div
        class="bg-black/30 border border-purple-500/20 rounded-lg p-0 overflow-hidden"
      >
        <div
          class="overflow-x-auto"
        >
          <table
            class="min-w-full text-sm"
          >
            <thead
              class="bg-gray-900"
            >
              <tr>
                <th
                  class="px-3 py-2 text-left"
                >
                  Time
                </th>
                <th
                  class="px-3 py-2 text-left"
                >
                  User
                </th>
                <th
                  class="px-3 py-2 text-left"
                >
                  Stage
                </th>
                <th
                  class="px-3 py-2 text-left"
                >
                  Allowed
                </th>
                <th
                  class="px-3 py-2 text-left"
                >
                  Model
                </th>
                <th
                  class="px-3 py-2 text-left"
                >
                  Command
                </th>
                <th
                  class="px-3 py-2 text-left"
                >
                  Reason
                </th>
                <th
                  class="px-3 py-2 text-left"
                >
                  Details
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="px-3 py-6 text-center text-gray-400"
                  colspan="8"
                >
                  No results
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="flex items-center justify-between p-3 bg-gray-900 border-t border-gray-800 text-xs"
        >
          <div>
            Total: 0
          </div>
          <div
            class="flex items-center gap-2"
          >
            <button
              class="px-2 py-1 rounded bg-gray-800 disabled:opacity-50"
              disabled=""
            >
              Prev
            </button>
            <div>
              Page 1 / 1
            </div>
            <button
              class="px-2 py-1 rounded bg-gray-800 disabled:opacity-50"
              disabled=""
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Override UI snapshots Editor shows "Using global" by default 1`] = `
<DocumentFragment>
  <div
    class="p-6 space-y-4 max-w-[1400px] mx-auto"
  >
    <div
      class="flex items-center justify-between"
    >
      <h1
        class="text-2xl font-semibold"
      >
        Agentic Editor (Split Pane)
      </h1>
      <div
        class="flex items-center gap-2"
      >
        <label
          class="flex items-center gap-1 text-sm"
        >
          <span>
            Model
          </span>
          <select
            class="border rounded px-2 py-1"
          >
            <option
              value="auto"
            >
              auto
            </option>
            <option
              value="gpt-4o"
            >
              gpt-4o
            </option>
            <option
              value="gpt-4o-mini"
            >
              gpt-4o-mini
            </option>
            <option
              value="gpt-3.5-turbo"
            >
              gpt-3.5-turbo
            </option>
            <option
              value="claude-3-5-sonnet"
            >
              claude-3-5-sonnet
            </option>
            <option
              value="local-llama"
            >
              local-llama
            </option>
          </select>
        </label>
        <span
          class="hidden md:inline text-xs text-gray-600"
        >
          Using global: 
          <b>
            auto
          </b>
        </span>
        <span
          class="text-xs px-2 py-1 rounded bg-gray-100 border"
        >
          Est. cost: $0.00000 (46 tok)
        </span>
        <label
          class="flex items-center gap-1 text-sm"
        >
          <input
            type="checkbox"
          />
           Expert mode
        </label>
        <button
          class="px-3 py-1 rounded bg-purple-600 text-white"
          title="Run preview"
        >
          Run
        </button>
        <button
          class="px-3 py-1 rounded bg-gray-200"
          title="Copy code"
        >
          Copy
        </button>
      </div>
    </div>
    <div
      class="border rounded relative select-none"
      style="height: calc(100vh - 180px);"
    >
      <div
        class="absolute top-0 left-0 h-full overflow-hidden"
        style="width: 50%;"
      >
        <div
          data-testid="monaco-mock"
        />
      </div>
      <div
        class="absolute top-0 h-full cursor-col-resize group bg-transparent"
        style="left: calc(50% - 3px); width: 6px;"
      >
        <div
          class="absolute top-1/2 -translate-y-1/2 left-1/2 -translate-x-1/2 w-1 h-24 rounded bg-gray-400 group-hover:bg-purple-500"
        />
      </div>
      <div
        class="absolute top-0 right-0 h-full overflow-auto"
        style="width: 50%;"
      >
        <div
          class="h-full p-3 space-y-3"
        >
          <div>
            <div
              class="text-sm text-gray-600 mb-2"
            >
              Output / Preview
            </div>
            <div
              class="border rounded p-3 bg-white overflow-auto"
              style="height: 220px;"
            >
              <pre
                class="text-xs whitespace-pre-wrap"
              >
                Ready.
              </pre>
            </div>
          </div>
          <div>
            <div
              class="text-sm text-gray-600 mb-2"
            >
              File Upload (OCR)
            </div>
            <div
              class="border-2 border-dashed rounded p-4 border-gray-300"
              role="button"
              tabindex="0"
            >
              <div
                class="text-xs text-gray-600"
              >
                Drop files here, or click to browse
              </div>
              <input
                class="hidden"
                multiple=""
                type="file"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="text-xs text-gray-600 flex items-center justify-between"
    >
      <span>
        Tip: Drag the vertical divider to resize panes. This split-pane lays the foundation for code editing workflows.
      </span>
      <div
        class="text-[11px] text-gray-500 flex items-center gap-2"
      >
        <span
          class="hidden md:inline"
        >
          Requests:
        </span>
        <span
          class="px-2 py-0.5 rounded bg-gray-900 border border-gray-700"
        >
          0
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
